src/GafferScene/Unencapsulate.cpp

```diff
@@ -163,29 +226,28 @@ void Unencapsulate::hashBranchObject( const ScenePath &parentPath, const ScenePa
 	}
 	else
 	{
-		h = capsule->scene()->objectHash( concatScenePath( capsule->root(), branchPath ) );
+		h = cs.scene()->objectPlug()->hash();
 	}
 }
 
 IECore::ConstObjectPtr Unencapsulate::computeBranchObject( const ScenePath &parentPath, const ScenePath &branchPath, const Gaffer::Context *context ) const
 {
-	IECore::ConstObjectPtr o = inPlug()->object( parentPath );
-	const Capsule *capsule = IECore::runTimeCast< const Capsule >( o.get() );
+	CapsuleScope cs( context, inPlug(), parentPath, branchPath, false );
  	if( branchPath.size() == 0 )
 	{
 		// We're at a root location, not inside a capsule.  If there's a capsule here, we return a null
 		// object ( the capsule has now been expanded, so we don't want it in the scene any more ).
 		// If there's no capsule, we're not expanding anything, just pass through the source scene
-		if( capsule )
+		if( cs.scene() )
 		{
 			return IECore::NullObject::defaultNullObject();
 		}
```

Comment:
Turns out there's an extra subtlety here - the capsule can have an object at its root, and we need to return that. Here's a little network to show you what I mean :

```
import Gaffer
import GafferScene
import IECore
import imath

Gaffer.Metadata.registerValue( parent, "serialiser:milestoneVersion", 0, persistent=False )
Gaffer.Metadata.registerValue( parent, "serialiser:majorVersion", 59, persistent=False )
Gaffer.Metadata.registerValue( parent, "serialiser:minorVersion", 0, persistent=False )
Gaffer.Metadata.registerValue( parent, "serialiser:patchVersion", 0, persistent=False )

__children = {}

__children["Sphere2"] = GafferScene.Sphere( "Sphere2" )
parent.addChild( __children["Sphere2"] )
__children["Sphere2"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Encapsulate1"] = GafferScene.Encapsulate( "Encapsulate1" )
parent.addChild( __children["Encapsulate1"] )
__children["Encapsulate1"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Unencapsulate2"] = GafferScene.Unencapsulate( "Unencapsulate2" )
parent.addChild( __children["Unencapsulate2"] )
__children["Unencapsulate2"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["PathFilter2"] = GafferScene.PathFilter( "PathFilter2" )
parent.addChild( __children["PathFilter2"] )
__children["PathFilter2"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Sphere2"]["__uiPosition"].setValue( imath.V2f( 57.0999985, 5.94999981 ) )
__children["Encapsulate1"]["in"].setInput( __children["Sphere2"]["out"] )
__children["Encapsulate1"]["filter"].setInput( __children["PathFilter2"]["out"] )
__children["Encapsulate1"]["__uiPosition"].setValue( imath.V2f( 57.0999985, 0.285937309 ) )
__children["Unencapsulate2"]["in"].setInput( __children["Encapsulate1"]["out"] )
__children["Unencapsulate2"]["filter"].setInput( __children["PathFilter2"]["out"] )
__children["Unencapsulate2"]["__uiPosition"].setValue( imath.V2f( 57.0999985, -5.37812519 ) )
__children["PathFilter2"]["paths"].setValue( IECore.StringVectorData( [ '/sphere' ] ) )
__children["PathFilter2"]["__uiPosition"].setValue( imath.V2f( 71.0477524, 5.50390577 ) )


del __children
```

Reply:
Fixed, and added test